simplifier/operations/interface.py

```diff
@@ -24,8 +24,11 @@ def add_operand(self, operand: WorldObject):
         super().add_operand(operand)
 
     def fold(self, keep_form: bool = False):
-        """Unary operations can not be folded."""
-        pass
+        """Fold a unary operation's operands."""
+        if isinstance(self.operand, Operation):
+            self.operand.fold()
+            if isinstance(self.operand, (AssociativeOperation, CommutativeOperation)):
+                self.operand._promote_single_operand()
```

Comment:
It seems a bit strange that we have to do this. When we simplify each operation separately. When we want to simplify the hole condition, it makes sense to assign it to a variable and when simplifying a variable, we do it in post order, so the operand should be simplified before the operation itself. And when simplifying the operand, we apply `_promote_single_operand`. Could you explain why we need this simplification?

Reply:
In the case of ```!((x != 0) | (x == 0))``` , the first unary, ```!``` would cause nothing to happen without this simplification, so the expression would not be simplified at all.

Reply:
I think when you define this condition as a variable, you can call `simplify` and it goes in post-order (https://github.com/fkie-cad/dewolf-logic/blob/37758b53152ccf678a84c948921e0bba814686e5/simplifier/world/nodes.py#L110). But, currently simplify does not use the RangeSimplifier. So we could either change it that it also use it. But, I think the following function should do what you want (not a good name)
```python
class BaseVariable:
......
    def simplify_all(self):
        from simplifier.operations import BitwiseNegate

        for node in list(self.world.iter_postorder(self)):
            if isinstance(node, BitwiseNegate):
                node.dissolve_negation()
        for node in list(self.world.iter_postorder(self)):
            if isinstance(node, Operation):
                RangeSimplifier.simplify(node)
```

Reply:
I've removed the original code in question to use the RangeSimplifier, hopefully it works better (?)